### PR TITLE
49172_modify_task_list_of_a_course

### PR DESCRIPTION
**Pull request by manazirahsan**

### CODE REVIEW

File: app/views/supervisor/course_subjects/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      <%= @course.name + " :: " + @subject.name %>
+    </div>
+  </div>
+  <div class="row">
+    <div class="col-md-8 col-md-offset-2">
+      <%= form_for [:supervisor, @course_subject], method: :patch do %>
+        <%= hidden_field_tag :course_id, @course.id %>
+        <%= hidden_field_tag :subject_id, @subject.id %>
+        <%= hidden_field_tag "course_subject[id]", @course_subject.id %>
+        <%= collection_check_boxes(:course_subject, :task_ids, @subject.tasks, :id, :name) do |b| %>
```
Comment (mdmahmudurrahman) [inline app/views/supervisor/course_subjects/edit.html.erb]: More than 80 characters!

File: app/views/supervisor/courses/show.html.erb [hunk 2]
```diff
@@ -52,6 +50,12 @@
         <tr>
           <td><%= link_to subject.name, [:supervisor, subject]  %></td>
           <td><%= subject.description.truncate 50 %></td>
+          <td>
+            <% unless @course.close? %>
+                <% course_subject = CourseSubject.find_by_course_id_and_subject_id(@course.id, subject.id) %>
+                <%= link_to t(:manipulate_task), edit_supervisor_course_subject_path(course_subject), class: "btn btn-info" %>
```
Comment (mdmahmudurrahman) [inline app/views/supervisor/courses/show.html.erb]: More than 80 characters!

File: app/controllers/supervisor/course_subjects_controller.rb [hunk 3]
```diff
@@ -1,46 +1,26 @@
...
-    @courses = Course.all
-  end
-
-  def load_subject
-    @subjects = Subject.all
-  end
-
-  def load_task
-    @tasks = Task.all
+    params.require(:course_subject).permit :id, task_ids: []
```
Comment (shakil-fg) [inline app/controllers/supervisor/course_subjects_controller.rb]: remove []

File: app/views/supervisor/course_subjects/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<% provide :title, t(:update_course)%>
+<div class="center">
+  <h1><%= t(:manipulate_task) %></h1>
```
Comment (shakil-fg) [inline app/views/supervisor/course_subjects/edit.html.erb]: remove ()

File: app/views/supervisor/course_subjects/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    </div>
+  </div>
+  <div class="row">
+    <div class="col-md-8 col-md-offset-2">
+      <%= form_for [:supervisor, @course_subject], method: :patch do %>
+        <%= hidden_field_tag :course_id, @course.id %>
+        <%= hidden_field_tag :subject_id, @subject.id %>
+        <%= hidden_field_tag "course_subject[id]", @course_subject.id %>
+        <%= collection_check_boxes(:course_subject, :task_ids, @subject.tasks, :id, :name) do |b| %>
+          <%= b.label(:"data-value" => b.value) { b.check_box + b.text } %><br>
```
Comment (shakil-fg) [inline app/views/supervisor/course_subjects/edit.html.erb]: don't use hash rocket

Comment (shakil-fg) [inline app/views/supervisor/course_subjects/edit.html.erb]: remove :

File: app/views/supervisor/courses/show.html.erb [hunk 6]
```diff
@@ -44,6 +41,7 @@
       <tr>
         <th><%= t (:subject_name) %></th>
```
Comment (shakil-fg) [inline app/views/supervisor/courses/show.html.erb]: remove ()

File: app/views/supervisor/courses/show.html.erb [hunk 7]
```diff
@@ -44,6 +41,7 @@
...
         <th><%= t (:subject_description) %></th>
```
Comment (shakil-fg) [inline app/views/supervisor/courses/show.html.erb]: remove ()

**Comment by shakil-fg** [comment]
OK

**Comment by mdmahmudurrahman** [comment]
OK

**Pull request merged**